### PR TITLE
Adds repeating DoAfters to Food and Drink

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR

Mainly this is a trial run to see if repeating doafters are a fit. If they work out fine with food and drink, they'll be added where necessary. Tools will have to wait for when the obsolete code gets cleaned up.

Cancelling the doafter will stop the repeat.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://user-images.githubusercontent.com/54602815/230751177-95854ddd-918a-4476-91ed-16da2449cf63.mp4

**Changelog**

:cl:
- add: The eating & drinking interactions now automatically repeat while there is still something left to consume.

### CODE REVIEW

File: Content.Shared/DoAfter/DoAfterArgs.cs [hunk 1]
```diff
@@ -127,6 +127,7 @@ public sealed class DoAfterArgs
     [DataField("blockDuplicate")]
     public bool BlockDuplicate = true;
 
+    //TODO: User perf to not cancel on second use on specific doafters
```
Comment (ElectroJr) [inline Content.Shared/DoAfter/DoAfterArgs.cs:130]: perf(ormance) -> pref(erence)

File: Content.Shared/DoAfter/DoAfterArgs.cs [hunk 2]
```diff
@@ -146,6 +147,14 @@ public sealed class DoAfterArgs
     public DuplicateConditions DuplicateCondition = DuplicateConditions.All;
     #endregion
 
+    //TODO: User perf to toggle repeat on specific doafters
+    /// <summary>
+    ///     If true, the doafter will automatically repeat until it's cancelled.
+    ///     Only set this in a do-after event that you would like repeated.
+    /// </summary>
+    [DataField("repeat")]
+    public bool Repeat = false;
```
Comment (ElectroJr) [inline Content.Shared/DoAfter/DoAfterArgs.cs:156]: This should just be moved to the do-after itself (i.e., `args.Args.Repeat -> args.Repeat`)

File: Content.Server/Nutrition/EntitySystems/FoodSystem.cs [hunk 3]
```diff
@@ -170,21 +171,24 @@ private void OnDoAfter(EntityUid uid, FoodComponent component, ConsumeDoAfterEve
             if (!TryGetRequiredUtensils(args.User, component, out var utensils))
                 return;
 
-            args.Handled = true;
+            var forceFeed = args.User != args.Target;
+
+            //If handled it won't repeat
```
Comment (ElectroJr) [inline Content.Server/Nutrition/EntitySystems/FoodSystem.cs:176]: out of date comment?

File: Content.Server/Nutrition/EntitySystems/DrinkSystem.cs [hunk 4]
```diff
@@ -377,11 +376,13 @@ private void OnDoAfter(EntityUid uid, DrinkComponent component, ConsumeDoAfterEv
             _reaction.DoEntityReaction(args.Args.Target.Value, solution, ReactionMethod.Ingestion);
             //TODO: Grab the stomach UIDs somehow without using Owner
             _stomachSystem.TryTransferSolution(firstStomach.Value.Comp.Owner, drained, firstStomach.Value.Comp);
-            args.Handled = true;
 
             var comp = EnsureComp<ForensicsComponent>(uid);
             if (TryComp<DnaComponent>(args.Args.Target, out var dna))
                 comp.DNAs.Add(dna.DNA);
+
+            if (!forceDrink && transferAmount > 0)
```
Comment (ElectroJr) [inline Content.Server/Nutrition/EntitySystems/DrinkSystem.cs:384]:
```suggestion
            if (!forceDrink && solution.Volume > 0)
```

**Review by ElectroJr — APPROVED** [review]
Other than changing the tranfer amount check, should be good?

**Comment by Buldinn** [comment]
This should be a toggle somewhere in the options or just straight removed.
You are forced to lay your drink down every time your character takes a sip or they will chug it forever.
Horrible for RP.

**Pull request merged**